src/vm/node_modules/VM.js

```diff
@@ -10897,12 +10897,29 @@ exports.reprovision = function (uuid, payload, options, callback)
             vs.once('ready', function (ready_ev) {
                 try {
                     vmobj = vminfodEventGetVM(uuid, ready_ev);
+
+                    if (!vmobj.zpool) {
+                        log.warn('failed to load \"zpool\" field, retrying '
```

Comment:
I think that OS-8316 fixed the underlying issue with vminfod here. This entire section can probably be reverted, at least until we can confirm that it's still ever an issue.